source/common/router/router.cc

```diff
@@ -656,7 +656,11 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
   callbacks_->streamInfo().setAttemptCount(attempt_count_);
 
   // Inject the active span's tracing context into the request headers.
-  callbacks_->activeSpan().injectContext(headers);
+  // If envoy will generate child span, it ignores to inject header and delegate it to upstream
```

Comment:
I think this needs a rephrase.  Would it be accurate to say

// If configured to generate a child spam, delegate injecting headers to the upstream request which generates the child span?

Then maybe it's worth having an integration test for tracing that we do get those headers injected, since we only have unit tests and this is verifying we only do it once across 2 separate components?

/wait

Reply:
It makes sense. It is not related directly to this PR. Thus I TODOed and will add in the future PR.